### DISCUSSION TITLE
Pagination in rest apis

### DISCUSSION THREAD

**Discussion by saikat056** [DependencyTrack/dependency-track#2157] [Q&A] [answered]
Hello Team,
 I am trying to paginate the following api in dependency_tracker. Can you please help me with the pagination details for this api? Thank you.
api/v1/component/project/<project_id>

I tried the options mentioned in this discussion: https://github.com/DependencyTrack/dependency-track/issues/209

It does not work for me.

**Comment by syalioune** [comment]
Hello @saikat056 
Can you please give some examples of requests you are trying and the expected results ?

**Comment by saikat056** [comment]
@syalioune 

If I hit api/v1/component/project/<project_id>, the api returns only 100 components although the project has almost 800 components. Can we paginate the list using offset and limit params?

For example, 

 ```
   api/v1/component/project/<project_id>?page=0&page_size=100
   api/v1/component/project/<project_id>?page=1&page_size=100
   api/v1/component/project/<project_id>?page=2&page_size=100
```

For now, there is no similar feature present. Thank you.

**Reply by nscuro** [reply]
Should be `pageSize`, not `page_size`.

**Comment by saikat056** [comment]
@nscuro I tried with both pageSize and page_size. It's not working. The api only returns first 100 components. All of the following apis are returning only first 100 components.

```
api/v1/component/project/<project_id>?pageSize=10
api/v1/component/project/<project_id>?pageSize=40
api/v1/component/project/<project_id>?pageSize=90
```

**Reply by nscuro** [reply]
`pageSize` and `pageNumber` only work when used together IIRC. The frontend uses this API using the `page*` parameters as well, so it's kind of odd that it doesn't work for you when directly interacting with it.

But glad that `offset` and `limit` work for you.

**Comment by saikat056** [comment, marked as answer]
@nscuro, @syalioune . Thank you for help. **offset** and **limit** params work for my case. 👍 
```
api/v1/component/project/<project_id>?offset=2&limit=1000
```
